libnetwork/networkdb/cluster.go

```diff
@@ -411,7 +411,7 @@ func (nDB *NetworkDB) reapTableEntries() {
 				return false
 			}
 
-			params := strings.Split(path[1:], "/")
+			params := strings.Split(string(path[1:]), "/")
 			nid := params[0]
 			tname := params[1]
 			key := params[2]
```

Comment:
Silly question (I can't comment on the line below); as the problem was that we were deleting _inside_ the loop, and as we're already taking a lock, would it make sense to _collect_ the list of entries to delete in `WalkPrefix`, and then delete them outside of it? Would that make it more performant?

Also wondering; it seems we're using the `WalkPrefix` to collect all entries with the given prefix so that we can delete all of them; if (IIUC), the format is `/<networkID>/<tableName>/<endpointID>/<value>`, but we only collect `/<networkID>/<tableName>/<endpointID>`, that means we're manually performing a delete multiple times (once for each `<value>`). Perhaps we could make use of `DeletePrefix()` 🤔

Reply:
Yeah I thought about this too. The current logic is very hairy because it's not simply a deleting a bunch of radixes, it also checks for each entry if it was successfully deleted and only then it updates the state of the number of entries in that network: https://github.com/moby/moby/pull/44501/files#diff-7b5363044492af3bcc6d8117470bfc51476d389c7a90670cfd5c62a83c32445aR777

And there's that whole timecompensation logic too. So yeah, again, I tried to change the least amount of logic in an area I'm not very comfortable in. If somebody wants to decipher the timecompensation logic and use DeletePrefix that's just an optimization which is not the purpose of this PR.